src/client.ts

```diff
@@ -91,8 +91,7 @@ export class SkynetClient {
       url = addUrlQuery(url, config.query);
     }
 
-    // No other headers.
-    const headers = config.customUserAgent && { "User-Agent": config.customUserAgent };
+    const headers = { ...config.headers, "User-Agent": config.customUserAgent && config.customUserAgent };
```

Comment:
still think this is dirty - it  yieds:
```
{
foo:"bar",
User-Agent:undefined
}
```
if `config.customUserAgent` is undefined.